src/fluree/db/query/exec/select.cljc

```diff
@@ -15,29 +16,8 @@
 
 #?(:clj (set! *warn-on-reflection* true))
 
-(defmulti display
-  "Format a where-pattern match for presentation based on the match's datatype.
-  Return an async channel that will eventually contain the formatted match."
-  (fn [match _compact]
-    (where/get-datatype-iri match)))
-
-(defmethod display :default
-  [match _]
-  (where/get-value match))
-
-(defmethod display const/iri-rdf-json
-  [match _compact]
-  (-> match where/get-value (json/parse false)))
-
-(defmethod display const/iri-id
-  [match compact]
-  (some-> match where/get-iri compact))
-
-(defmethod display const/iri-vector
-  [match _compact]
-  (some-> match where/get-value vec))
-
 (defprotocol ValueSelector
+  :extend-via-metadata true
```

Comment:
I think this namespace is convoluted by relying on this, and I don't see any benefit.

I think instead, there should be two additional namespaces: fluree.db.query.exec.select.fql, and fluree.db.query.exec.select.sparql for the different output formats, each with their own implementations of the `ValueSelector` protocol that don't reference each other.

Then, during parse time, you can read the output format once and decide which `ValueSelector` records to instantiate based on the select clause.

Reply:
I chose `:extend-via-metadata` because the other protocols the records implement (`ValueAdapter`, `SolutionModifier`) aren't affected by the display and if I create separate `Selector` records for FQL vs SPARQL I'd have to move the common implementations to a third, common namespace or duplicate them, and on the balance it seemed that `:extend-via-metadata` was the simpler choice.

Reply:
Ok. I'm worried about the performance implications of extending via metadata as well as the semantic implications of having opaque functions hanging off of records. 

This pr has been held open for long enough so I'll let this through and we can address these concerns later if they prove to be a problem.